GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -163,6 +163,11 @@ protected override void Dispose(bool disposing)
                 // Make sure to mark the background queue as complete before disposing the cancellation token sequence.
                 _backgroundQueue.Complete();
                 _backgroundCancellationSequence.Dispose();
+                if (!_backgroundCancellationToken.IsCancellationRequested)
+                {
+                    _backgroundCancellationSequence.CancelCurrent();
```

Comment:
❔ Is it a good idea / really necessary to call `CancelCurrent` after `Dispose`?

Reply:
Not strictly necessary, the core problem for the tests were that artificial commits were added without any other commits caused `OnRevisionReadCompleted()` to take the quick path and never quit the loading phase, background thread would never finish.
So I believe it should be included.

Reply:
IIUIC, not necessary (and not critical) since `Dispose` does its job (and `CancelCurrent` will not throw after `Dispose` but should not be called because it creates a new token):
```
        /// <summary>
        /// Cancels the current token and disposes this object.
        /// </summary>
        /// <remarks>
        /// Only the first call to this method per instance of this class has any effect.
        /// </remarks>
        public void Dispose()
        {
            var cancellationTokenSource = Interlocked.Exchange(ref _cancellationTokenSource, null);

            if (cancellationTokenSource is null)
            {
                return;
            }

            cancellationTokenSource.Cancel();
            cancellationTokenSource.Dispose();
        }
```

Reply:
When I debugged, the cancellationToken in RunBackgroundAsync() was never cancelled. So something is not working here.
But the core problem was that the artificial commits were added starting the background queue without the queue being stopped. That is a little error prone. But #8809 is submitted to fix this - should be completed.

So I will revert this.